### ISSUE TITLE
Feature Request: Allow list for certain urls or url parameters?

### ISSUE THREAD

**Issue by nextgenthemes** [bhadaway/stop-spammers#50]
The aggressiveness of this plugin caused a lot of false positive on API calls and other things where I think I could just allow everything with certain urls/args because bots would not know about them.

I had this issue https://wordpress.org/support/topic/access-denied-clashes-with-edd-download-license-checking/ as well.

**Comment by bhadaway** [comment]
Yes, this plugin is very aggressive. In fact, that's stated directly in its description. Because of this, it's probably not the best option for commercial sites that can't risk even temporarily blocking good bots or users.

You can already allow or block at the IP level, which I think is probably better than using a URL anyways. Whatever services are getting blocked, they should provide a list of public IPs they use in which you can whitelist. If you can provide the names of the services that are getting blocked, I can help you track down that info.

Concerning any conflicts with other plugins, going to _Stop Spammers_ > _Protection Options_ > _And checking the box for "Only Use the Plugin for Standard WordPress Forms"_ will solve most of those issues.

**Comment by nextgenthemes** [comment]
I know that it says aggressive in the description, can be a good thing if there are ways to fine tune.

No IP allowing does not make sense at all in this case as this calls come from random WordPress sites. My customers sites. Allowing certain urls and parameters (that trigger api calls where bots would not even be able to to harm anyway) would fix this issues and commercial site could just add those needed and still sue this plugin.

I will check that option, did not know about it, but I have forms in place that are not standard so just assuming this will work it will also allow spam through those forms.

**Comment by bhadaway** [comment]
Apologies if I'm misunderstanding.

I would think the IPs of the service in which is providing the API would be all that's needed to be whitelisted, but actual end-user/end-user site IPs are getting blocked? I'm not sure I understand the issue and this might beyond the scope of the plugin.

But, is there any chance you can explain in more detail what service/feature you're providing that's getting jammed up and possibly a link to your site and the exact steps needed to recreate the issue?

I'm usually able to offer a creative fine-tuning solution once I understand the issue better.

Thanks

**Comment by nextgenthemes** [comment]
Its actually WP sites doing API calls for EDD Software licensing. They call example.com/edd-sl/ and example.com?edd_listener something like that. Its wp_remote_get calls and they expect data from those calls but this plugins blocks them.

You actually to not really need to understand this. WP has APIs for url params and the path can also be detected so I think there could be textareas one for "allowed paths" and one for "allowed parameters" where each line is another exclude rule. Or a filter for those so certain pages just get excluded and will be untouched by this plugin.

I for now use the option you suggested but there are for example login and registration forms EDD creates that may not be considers "standard forms". I not yet looked into how this plugin defines those.

**Comment by bhadaway** [comment]
You're right, this should probably be something added to the Allow Lists page for allowing WooCommerce, Easy Digital Downloads, and similar plugins to operate certain functions untouched by Stop Spammers.

However, that's probably outside the scope of development at the moment. I'll at least tag this issue for enhancement and help wanted.

For now, I'm sure we can figure out a solution for you specifically though. So, checking "Only Use the Plugin for Standard WordPress Forms" was enough to stop the EDD functions from failing? Is spam now getting in because of this? If so, we can probably refine blocking for the specific spam you're dealing with.

**Comment by nextgenthemes** [comment]
So far I have not gotten any spam so looks like I not really need it. I will report back if I do. I may have a look at the code if feel I need it and find some time. If the option is unset it seems that this plugin hooks into all requests so I think it should not be that hard to do this.

**Comment by bhadaway** [comment]
Sounds good. :)

**Comment by nextgenthemes** [comment]
Actually another thought. I think you should not enable that option by default. You should probably tell people about that is blocks **everything** the seems automated everywhere on the site. And that they can make the plugin even more aggressive my enabling that option. I do not this its just EDD and Woocommerce, there are a lot of things like plugins migrating databases ... this option is very risky and can mean a lot of pain to people so I do not think its a good fit to enable it by default.
